apps/dashr-tsne/app.R

```diff
@@ -0,0 +1,1195 @@
+)
+
+datLabels <<- l.(
+  mnist_3000 = unique(fread("data/mnist_3000_labels.csv", header=T))
+  , 
+  fashion_3000= unique( fread("data/fashion_3000_labels.csv", header=T))
+  ,
+  cifar_gray_3000 = unique(fread("data/cifar_gray_3000_labels.csv", header=T))
+  ,
+  wikipedia_3000 = unique(fread("data/wikipedia_3000.csv", header=T)[, 1])
+  ,
+  crawler_3000 = unique(fread("data/crawler_3000.csv", header=T)[, 1])
+  ,
+  twitter_3000 =  unique(fread('data/twitter_3000.csv', encoding='Latin-1', header=T)[,1])
+)
+
+app$callback(
+  output = l.(id='div-wordemb-controls', property='style')
+  ,
+  params = l.(input(id='dropdown-dataset', property='value'))
+  ,
+  function(dataset){
+    if(is.element(dataset, WORD_EMBEDDINGS)){
+      return(l.(width='30vh', display='block'))
+    } else {
+      return(l.(display='none'))
+    }
+  }
+)
+#
+app$callback(
+  output = l.(id='dropdown-word-selected', property='disabled')
+  ,
+  params = l.(input(id='radio-wordemb-display-mode', property='value'))
+  ,
+  function(mode){
+    if (mode == 'neighbors'){
+      return(FALSE)
+    } else {
+      return(TRUE)
+    }
+  }
+  
+)
+#
+#helper function
+labeledList <- function(n){
+  l <- l.()
+  l$label = n
+  l$value = n
+  return(l)
+}
+
+#
+app$callback(
+  output = l.(id='dropdown-word-selected', property='options')
+  ,
+  params = l.(input(id='dropdown-dataset', property='value'))
+  ,
+  function(dataset){
+    
+    D <- datLabels[[dataset]]
+    L <- lapply(unlist(D), labeledList)
+    
+    return(unname(L))
+    
+    
+    
+  }
+)
+
+#############CALLBACK that shows 3D plot#####
+
+
+
+## helper function
+listedDTs <- function(n, dt){
+  return(dt[dt$label==n])
+}
+#
+### Euclidean distance
+eu.norm <- function(v1, v2){
+  return(sqrt(sum((v1-v2)^2)))
+}
+#
+### functions that enable us to load
+#
+generate_figure_word_vec <- function(embedding_df, wordemb_display_mode, selected_word, datTbl){
+  
+  colnames(embedding_df) <- c("label", 'x', 'y', 'z')
+  
+  #Regular displays the full scatter plot with only circles
+  if(wordemb_display_mode == 'regular'){
+    plot_mode = 'markers'
+    
+    p <- plot_ly(embedding_df, type='scatter3d', x=~x, y=~y, z=~z, color = '#ED9C69',text=~label, mode='markers', marker=list(symbol='circle', size=2.5))
+    
+    return(p)
+  }
+  # Nearest Neighbors display only the 100 nearest neighbors of the selected_word, in text rather than circles
+  else if(wordemb_display_mode == 'neighbors'){
+    if(selected_word == ''){
+      return(plot_ly(type =  'scatter3d', x=0, y=0, z=0, mode='text', text='NA'))
+    }
+    plot_mode = 'text'
+    #Get the nearest neighbors incides using Euclidean distance
+    
+    
+    
+    
+    selected_vector = filter(datTbl, label==selected_word)
+    mtx <- as.matrix(datTbl[, -1])
+    sel_vec <- unlist(selected_vector[, -1])
+    
+    distances <- mtx %>% apply(., 1, . %>% eu.norm(., sel_vec))
+    datTbl$distance = distances
```

Comment:
Any reason why you don't just do:
```
datTbl$distance <- mtx %>% 
  apply(., 1, . %>% eu.norm(., sel_vec))
```

Or something along those lines?

Reply:
Also, probably best to stick to `<-` for assignment instead of `=`, both for consistency and to adhere to google's R style guidelines (should be replaced throughout the code where there is `=` for assignment)